Mapping/SolARPipeline_Mapping_Mono/tests/SolARPipelineTest_Mapping_Mono/SolARPipelineTestMappingMono_main.cpp

```diff
@@ -115,12 +115,12 @@ int main(int argc, char ** argv)
 			}
 		}		
 		
-		std::vector<SRef<CloudPoint>> pointCloud;
-		std::vector<Transform3Df> keyframePoses;
-		if (gMappingPipeline->getDataForVisualization(pointCloud, keyframePoses) == FrameworkReturnCode::_SUCCESS) {
-			LOG_INFO("Number of cloud points: {}", pointCloud.size());
-			LOG_INFO("Number of keyframes: {}", keyframePoses.size());
-			while (g3DViewer->display(pointCloud, keyframePoses[keyframePoses.size() - 1], keyframePoses) == FrameworkReturnCode::_SUCCESS);
+		while (true) {
+			std::vector<SRef<CloudPoint>> pointCloud;
+			std::vector<Transform3Df> keyframePoses;
+			if ((gMappingPipeline->getDataForVisualization(pointCloud, keyframePoses) != FrameworkReturnCode::_SUCCESS) ||
```

Comment:
I find it odd (from an API design stand point) to return an ERROR when there is no data.
Error should be used if there was actually an error while executing the code, but if there's no data, one can only check pointCloud size, or am I missing something? (is ERROR a different case?) 
Why guard display with a check on getDataForVisualization() return code ? Can't display() be called on an empty vector ? If so, it may be implemented like:

```
do
{
    std::vector<SRef<CloudPoint>> pointCloud;
    std::vector<Transform3Df> keyframePoses;
    gMappingPipeline->getDataForVisualization(pointCloud, keyframePoses);
}
while ( g3DViewer->display(pointCloud, keyframePoses[keyframePoses.size() - 1], keyframePoses)
           != FrameworkReturnCode::_STOP) )
```

Don't necessarily change yours, I just found it might be simpler to read (if I didn't make a mistake, I didn't test it)